src/main/kotlin/no/nav/syfo/domain/Varsel.kt

```diff
@@ -0,0 +1,21 @@
+package no.nav.syfo.domain
+
+import no.nav.syfo.util.nowUTC
+import java.time.OffsetDateTime
+import java.util.*
+
+data class Varsel private constructor(
+    val uuid: UUID,
+    val document: List<DocumentComponent>,
+    val createdAt: OffsetDateTime,
+    val journalpostId: String?,
+) {
+    companion object {
+        fun create(document: List<DocumentComponent>) = Varsel(
```

Comment:
Hvor kommer egentlig dette mønsteret fra? Hvorfor har vi ikke bare konstruktør? (Tror jeg spurte om det i en annen PR en gang, men finner det ikke igjen 🤷‍♂️)

Reply:
Snakket litt med Anders om det! Tanken er vel at domeneobjektet selv skal ha oversikt over lovlige tilstander for feks initiering. Det kalles [factory-funksjoner](https://blog.kotlin-academy.com/item-30-consider-factory-functions-instead-of-constructors-e1c747fc475) visstnok (jeg leste ikke hele denne, bare så på overskriftene, hehe).

Det _kan_ man jo gjøre med ulike konstruktører også, men på denne måten har man kanskje en liten fordel ved at man får med et funksjonsnavn som forteller i større grad hva som gjøres. Det er lettere å bruke `Vurdering.createForhandsvarsel()` enn å velge ut riktig konstruktør for forhåndsvarselet ved `Vurdering()`.